clients/java/signalr/src/main/java/com/microsoft/aspnet/signalr/HubConnection.java

```diff
@@ -32,10 +35,37 @@
     private Map<String, String> headers = new HashMap<>();
     private ConnectionState connectionState = null;
     private HttpClient httpClient;
+    private Timer pingTimer = null;
+    private long nextServerTimeout = 0;
+    private long nextPingActivation = 0;
+    private long keepAliveInterval = 15000;
+    private long serverTimeout = 30000;
+    private long tickRate = 1000;
 
     private static ArrayList<Class<?>> emptyArray = new ArrayList<>();
     private static int MAX_NEGOTIATE_ATTEMPTS = 100;
 
+    public void setServerTimeoutInMilliseconds(long serverTimeout) {
+        this.serverTimeout = serverTimeout;
+    }
+
+    public long getServerTimeoutInMilliseconds() {
+        return this.serverTimeout;
+    }
+
+    public void setKeepAliveIntervalInMilliseconds(long keepAliveInterval) {
```

Comment:
Can we use [Duration](https://docs.oracle.com/javase/8/docs/api/java/time/Duration.html) so that we don't have to specify the units in the name? If we have to use `long`s, then I agree that the units should be here but I'd like to avoid `camelCaseSoupWithABunchOfWordsInIt`